Realm/RLMMongoCollection.mm

```diff
@@ -0,0 +1,307 @@
+                                                  realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        completion((NSArray<NSDictionary<NSString *, id<RLMBSON>> *> *)RLMConvertBsonToRLMBSON(*documents), nil);
+    });
+}
+
+- (void)findWhere:(NSDictionary<NSString *, id<RLMBSON>> *)document
+       completion:(RLMMongoFindBlock)completion {
+    [self findWhere:document options:[[RLMFindOptions alloc] init] completion:completion];
+}
+
+- (void)findOneDocumentWhere:(NSDictionary<NSString *, id<RLMBSON>> *)document
+                     options:(RLMFindOptions *)options
+                  completion:(RLMMongoFindOneBlock)completion {
+    [self collection:self.name].find_one(static_cast<realm::bson::BsonDocument>(RLMConvertRLMBSONToBson(document)),
+                                         [options _findOptions],
+                                         [completion](realm::util::Optional<realm::bson::BsonDocument> document,
+                                                      realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        completion((NSDictionary<NSString *, id<RLMBSON>> *)RLMConvertBsonToRLMBSON(*document), nil);
+    });
+}
+
+- (void)findOneDocumentWhere:(NSDictionary<NSString *, id<RLMBSON>> *)document
+                  completion:(RLMMongoFindOneBlock)completion {
+    [self findOneDocumentWhere:document options:[[RLMFindOptions alloc] init] completion:completion];
+}
+
+- (void)insertOneDocument:(NSDictionary<NSString *, id<RLMBSON>> *)document
+               completion:(RLMMongoInsertBlock)completion {
+    [self collection:self.name].insert_one(static_cast<realm::bson::BsonDocument>(RLMConvertRLMBSONToBson(document)),
+                                           [completion](realm::util::Optional<realm::ObjectId> objectId,
+                                                        realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        completion([[RLMObjectId alloc] initWithValue:*objectId], nil);
+    });
+}
+
+- (void)insertManyDocuments:(NSArray<NSDictionary<NSString *, id<RLMBSON>> *> *)documents
+                 completion:(RLMMongoInsertManyBlock)completion {
+    [self collection:self.name].insert_many(static_cast<realm::bson::BsonArray>(RLMConvertRLMBSONToBson(documents)),
+                                            [completion](std::vector<realm::ObjectId> insertedIds,
+                                                         realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        NSMutableArray *insertedArr = [[NSMutableArray alloc]  init];
```

Comment:
`[[NSMutableArray alloc] initWithCapacity:insertedIds.size()]` is an easy minor optimization.